model/localisation.go

```diff
@@ -1,8 +1,29 @@
 package model
 
-// Localisation data
+import (
+	"github.com/ONSdigital/dp-renderer/helper"
+)
+
+/* Localisation offers text or a localised substitute.
+Text is displayed as-is.
+LocaleKey is a key into the toml files found in assets/locales.
+Plural chooses the singular or plural form of the phrase selected by LocaleKey.
+*/
 type Localisation struct {
 	Text      string `json:"text"`
 	LocaleKey string `json:"locale_key"`
 	Plural    int    `json:"plural"`
 }
+
+// Localise by preference when a LocaleKey is provided, defaulting to Text otherwise.
+func (localisation Localisation) Localise(language string) string {
```

Comment:
Few comments about this helper func...

- This isn't the right location for a helper function, it should be in the `helper` package
- There are no unit tests
- You're essentially inheriting the `localise` helper function with this which you've made possible but as it's not an object orientated language it's not [idiomatic of the language](https://flaviocopes.com/golang-is-go-object-oriented/). I'm not anti your suggestion but I think it should be either an extension to the existing `localise` method or a new method either option should be backed up with appropriate unit tests

Reply:
I'm not sure why you've linked to an article that declares Go 'an object orientated language in a refreshing and sane way' in support of your assertion that Go isn't an object orientated language.

There is precedence for methods in the Pagination model - I can rename the method if there's a concern that this masks or could be mistaken for the existing helper.Localise()

Units tests I can add

Reply:
`helper.Localise` is a function from a package of such functions, and calling it should no more be conflated with the concept of inheritance than calling any other package function, like `math.Round`. The method being similarly named to the package function should not bestow upon the call an intention to create inheritance where no such intention exists.

The methods in the Pagination model were, in turn, inspired by Go code present in the estate that uses this approach - I offer the Service struct in the frontend controllers as an example. Supplying methods in this manner is also ably demonstrated by the article you linked to.

Reply:
Having achieved concensus in the frontend guild meeting the model methods have been updated with a `Func` prefix, in light of the limitions of the Go Template syntax in highlighting the invocation of a method rather than a reference to the property of a model.

Reply:
Having achieved concensus in the frontend guild meeting the model methods have been updated with a `Func` prefix, in light of the limitions of the Go Template syntax in highlighting the invocation of a method rather than a reference to the property of a model.